op-service/client/dial_test.go

```diff
@@ -0,0 +1,17 @@
+package client
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestIsURLAvailable(t *testing.T) {
+	go func() {
+		_ = http.ListenAndServe(":8989", nil)
+	}()
+
+	require.True(t, IsURLAvailable("http://localhost:8989"))
+	require.False(t, IsURLAvailable("http://localhost:9898"))
```

Comment:
For the false case maybe use an IPv6 black-hole IP address instead? Then the dial will always fail.

Reply:
Using port 0 instead.